src/front/app/features/challenge/components/Challenge.js

```diff
@@ -78,7 +85,14 @@ class Challenge extends Component {
 
 Challenge.propTypes = {
     challenge: ImmutablePropTypes.map.isRequired,
+    user: ImmutablePropTypes.map,
+    getUser: PropTypes.func.isRequired,
+    userId: PropTypes.string,
     getChallenge: PropTypes.func.isRequired,
+    getAcceptedChallenges: PropTypes.func.isRequired,
```

Comment:
We can avoid retrieving wish list and accepted challenges by moving related bunch of logic to be side